src/Microsoft.Extensions.WebEncoders/EncoderServiceCollectionExtensions.cs

```diff
@@ -17,14 +20,23 @@ public static class EncoderServiceCollectionExtensions
         /// </summary>
         /// <param name="services">The <see cref="IServiceCollection"/>.</param>
         /// <returns>The <see cref="IServiceCollection"/> instance after the encoders have been added.</returns>
-        public static IServiceCollection AddWebEncoders(this IServiceCollection services)
+        public static void AddWebEncoders(this IServiceCollection services)
```

Comment:
Rather than removing chaining, couldn't we just name the builder methods different than Add?  Build/RegisterMvc

```
    services.AddSingleton<Fooey>()
       .BuildMvc(o => o.Stuff = "yesh")
       .MvcSpecificStuff();
```

RE enforcement about builders being IServiceCollections, I'm not sure how it would be any different than the manual code review/due diligence we have now